scvi/metrics/classification.py

```diff
@@ -32,38 +32,38 @@ def compute_accuracy_tuple(y, labels):
 
 @no_grad()
 @eval_modules()
-def compute_accuracy(vae, data_loader, classifier=None):
+def compute_accuracy(vae, data_loader, classifier=None, use_cuda=True):
     all_y_pred = []
     all_labels = []
 
     for i_batch, tensors in enumerate(data_loader):
-        if vae.use_cuda:
-            tensors = to_cuda(tensors)
+        tensors = to_cuda(tensors, use_cuda=use_cuda)
         sample_batch, _, _, _, labels = tensors
         sample_batch = sample_batch.type(torch.float32)
         all_labels += [labels.view(-1)]
 
-        if classifier is not None:
-            # Then we use the specified classifier
-            mu_z, _, _ = vae.z_encoder(sample_batch)
-            y_pred = classifier(mu_z).argmax(dim=-1)
-        else:
-            # Then the vae must implement a classify function
+        if hasattr(vae, 'classify'):
```

Comment:
Any way to avoid branching on `hasattr`? It's kind of complicated. Or may be there's no other way....

Reply:
agreed ! this is certainly bad design. If `.accuracy` was a function of the inference object, there wouldn't be any branching. Probably better to do it that way !